piker/data/_sampling.py

```diff
@@ -158,36 +162,40 @@ async def iter_ohlc_periods(
     '''
     # add our subscription
     subs = sampler.subscribers.setdefault(delay_s, [])
-    subs.append(ctx)
+    await ctx.started()
+    async with ctx.open_stream() as stream:
+        subs.append(stream)
 
-    try:
-        # stream and block until cancelled
-        await trio.sleep_forever()
-    finally:
         try:
-            subs.remove(ctx)
-        except ValueError:
-            log.error(
-                f'iOHLC step stream was already dropped for {ctx.chan.uid}?'
-            )
+            # stream and block until cancelled
+            await trio.sleep_forever()
+        finally:
+            try:
+                subs.remove(stream)
+            except ValueError:
+                log.error(
+                    f'iOHLC step stream was already dropped {ctx.chan.uid}?'
+                )
 
 
 async def sample_and_broadcast(
 
     bus: '_FeedsBus',  # noqa
     shm: ShmArray,
     quote_stream: trio.abc.ReceiveChannel,
+    brokername: str,
     sum_tick_vlm: bool = True,
 
 ) -> None:
 
     log.info("Started shared mem bar writer")
 
+    overruns = Counter()
```

Comment:
when the data feed is pushing data faster then the consumer side is pulling it; it's standard real-time queueing parlance.
in this case this is a counter of how many overruns take place before we bail on the push side of the memory channel/stream, since usually it would mean the consumer died and we're pushing to noone.

Reply:
ah thanks. So this helps us counter how many overruns we got so we can kill the stream once we hit a certain threshold?